bm-persona/Dining/DiningDetailViewController.swift

```diff
@@ -0,0 +1,496 @@
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.adjustsFontSizeToFitWidth = true
+        label.minimumScaleFactor = 0.75
+        label.numberOfLines = 1
+        return label
+    }()
+}
+
+extension DiningDetailViewController {
+    
+    func setUpOverviewCard() {
+        view.layoutMargins = UIEdgeInsets(top: kViewMargin, left: kViewMargin, bottom: kViewMargin, right: kViewMargin)
+        
+        view.addSubview(card)
+        card.topAnchor.constraint(equalTo: view.layoutMarginsGuide.topAnchor).isActive = true
+        card.leftAnchor.constraint(equalTo: view.layoutMarginsGuide.leftAnchor).isActive = true
+        card.rightAnchor.constraint(equalTo: view.layoutMarginsGuide.rightAnchor).isActive = true
+        card.heightAnchor.constraint(equalToConstant: 200).isActive = true
+        view.layoutSubviews()
+        
+        card.addSubview(clockIcon)
+        card.addSubview(openTimeLabel)
+        card.addSubview(openTag)
+        card.addSubview(chairImage)
+        card.addSubview(capBadge)
+        card.addSubview(faveButton)
+        card.addSubview(diningImage)
+        card.addSubview(phoneIcon)
+        card.addSubview(phoneLabel)
+        card.addSubview(distIcon)
+        card.addSubview(distLabel)
+        card.addSubview(addressIcon)
+        card.addSubview(addressLabel)
+        card.addSubview(nameLabel)
+        
+        clockIcon.bottomAnchor.constraint(equalTo: card.layoutMarginsGuide.bottomAnchor).isActive = true
+        clockIcon.leftAnchor.constraint(equalTo: card.layoutMarginsGuide.leftAnchor).isActive = true
+        clockIcon.widthAnchor.constraint(equalToConstant: 16).isActive = true
+        clockIcon.heightAnchor.constraint(equalToConstant: 16).isActive = true
+        
+        let formatter = DateIntervalFormatter()
+        formatter.dateStyle = .none
+        formatter.timeStyle = .short
+        let date = Date()
+        openTimeLabel.text = "     "
+        if let intervals = diningHall!.weeklyHours?.hoursForWeekday(DayOfWeek.weekday(date)) {
+            openTimeLabel.text = "Closed Today"
+            var nextOpenInterval: DateInterval? = nil
+            for interval in intervals {
+                if interval.contains(date) {
+                    nextOpenInterval = nil
+                    openTimeLabel.text = formatter.string(from: interval.start, to: interval.end)
+                    break
+                } else if date.compare(interval.start) == .orderedAscending {
+                    if nextOpenInterval == nil {
+                        nextOpenInterval = interval
+                    } else if interval.compare(nextOpenInterval!) == .orderedAscending {
+                        nextOpenInterval = interval
+                    }
+                }
+            }
+            if nextOpenInterval != nil {
+                openTimeLabel.text = formatter.string(from: nextOpenInterval!.start, to: nextOpenInterval!.end)
+            }
+        }
```

Comment:
The intervals should be in sorted order, but this works too, so I guess it's fine.